cli/client.go

```diff
@@ -2050,3 +2050,61 @@ func logEntryFieldsToString(fields []*structpb.Struct) (string, error) {
 	}
 	return message + "}", nil
 }
+
+type readOAuthAppArgs struct {
+	OrgID    string
+	ClientID string
+}
+
+const (
+	clientAuthenticationPrefix = "CLIENT_AUTHENTICATION_"
+	pkcePrefix                 = "PKCE_"
+	urlValidationPrefix        = "URL_VALIDATION_"
+	enabledGrantPrefix         = "ENABLED_GRANT_"
+)
+
+// ReadOAuthAppAction is the corresponding action for 'organizations auth-service oauth-app read'.
+func ReadOAuthAppAction(c *cli.Context, args readOAuthAppArgs) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+
+	return client.readOAuthAppAction(c, args.OrgID, args.ClientID)
+}
+
+func (c *viamClient) readOAuthAppAction(cCtx *cli.Context, orgID, clientID string) error {
+	if err := c.ensureLoggedIn(); err != nil {
+		return err
+	}
+
+	req := &apppb.ReadOAuthAppRequest{OrgId: orgID, ClientId: clientID}
+	resp, err := c.client.ReadOAuthApp(c.c.Context, req)
+	if err != nil {
+		return err
+	}
+
+	config := resp.OauthConfig
+	printf(cCtx.App.Writer, "OAuth config for client ID %s:", clientID)
+	printf(cCtx.App.Writer, "")
+	printf(cCtx.App.Writer, "Client Authentication: %s", protoToString(config.ClientAuthentication.String(), clientAuthenticationPrefix))
```

Comment:
Do you know why we return these prefixes if we are going to be removing them? I know this isn't your problem but that is the case we should be doing this formatting in the server itself

Reply:
The prefix is there for all proto enums. In the Go SDK, we put these enums into shadow types with prefixes as well because `funcCall(passwordRequired, methodUnspecified)` is more readable than `funcCall(required, unspecified ...)`. In this case, it's a little annoying to present, but I think it's good practice to keep these as is since the configs are returning specific enum values.

Reply:
As a general rule, we're avoiding giving users direct access to proto types, so we define shadow types that wrap around proto types. There are a lot of examples in the app wrappers (i.e. [FragmentVisibility](https://github.com/viamrobotics/rdk/blob/5d987fddb6dce7fb5c66191452504b7bc2c21185/app/app_client.go#L168-L180)). Then we have [helper functions](https://github.com/viamrobotics/rdk/blob/5d987fddb6dce7fb5c66191452504b7bc2c21185/app/app_client.go#L2101-L2127) to convert them.

Reply:
You'll see that keep these prefixes as it's a good idea to have an inkling of what `public` means when it's used. Which is why in this case when we're trying to present them, it's a little annoying.